src/library/cratehighlightdelegate.cpp

```diff
@@ -0,0 +1,45 @@
+#include "library/cratehighlightdelegate.h"
+#include "library/treeitem.h"
+#include "trackinfoobject.h"
+#include "libraryfeature.h"
+#include "cratefeature.h"
+
+CrateHighlightDelegate::CrateHighlightDelegate(QObject* parent)
+    : QStyledItemDelegate(parent) {
+}
+
+// This will be called to by Qt before painting the TreeView-Item. Set up styles here
+void CrateHighlightDelegate::initStyleOption(QStyleOptionViewItem* option,
+                                             const QModelIndex& index) const {
+    if (!index.isValid()) {
+        return;
+    }
+
+    QStyledItemDelegate::initStyleOption(option, index);
+    QStyleOptionViewItemV4 *optionV4 = qstyleoption_cast<QStyleOptionViewItemV4*>(option);
+
+    // If the item has no parent then it is a top-level sidebar item and its
+    // internalPointer is of type SidebarModel*, not TreeItem*.
+    if (!index.parent().isValid()) {
+        return;
+    }
+
+    TreeItem* item = static_cast<TreeItem*>(index.internalPointer());
+    if (item == NULL) {
+        return;
+    }
+
+    LibraryFeature* pFeature = item->getFeature();
+    if (pFeature == NULL) {
+        return;
+    }
+
+    TrackPointer pTrack = pFeature->getSelectedTrack();
+    if (pTrack.isNull()) {
+        return;
+    }
+
+    if (pFeature->isTrackInChildModel(pTrack->getId(), item->dataPath())){
```

Comment:
It's a little tricky due to the structure of the delegate -- but yes that would be better. 

This could be done without a delegate by changing TreeItemModel::data() to return a bolded font for the Qt::FontRole if the crate is selected. That way you could get the list of crates/playlists that should be bold there and then access the list in TreeItemModel::data directly.